src/providers/vmware/amd64.rs

```diff
@@ -4,9 +4,14 @@
 
 use super::VmwareProvider;
 use anyhow::{bail, Context, Result};
+use base64::{engine::general_purpose, Engine as _};
+use libflate::gzip::Decoder;
+use std::io::Read;
 
 /// Guestinfo key for network kargs.
 static INITRD_NET_KARGS: &str = "guestinfo.afterburn.initrd.network-kargs";
+static METADATA: &str = "guestinfo.metadata";
+static METADATA_ENCODING: &str = "guestinfo.metadata.encoding";
```

Comment:
Hmm, I'm unsure about the use of `guestinfo.metadata`. The PR description says:

> Since the format should not depend on whether the VM runs
uses Ignition and Afterburn or Cloud-Init, the idea is to also support
the guestinfo.metadata variable as used by Cloud-Init which contains
Netplan YAML network configuration.

But in the cloud-init case, users would normally put their netplan configs under e.g. `write_files` or something, right? I'm worried re-using the same key as cloud-init on systems with Ignition and Afterburn where one would've expected an Ignition config to live there instead. (Another way to say this: either that key should be read by Ignition or not at all.) I'm not sure on the historical reason why Ignition created its own key rather than conform to the established `guestinfo.metadata` key, but at this point I think it'd add more to the confusion to introduce a reader.

If the goal is to try to land on something that could potentially be implemented cross-provisioning tool, then let's create a new key. E.g. `guestinfo.netplan.metadata`? Or maybe `guestinfo.networking.metadata` to make it agnostic to the networking tool too since one could imagine feeding nmstate configs through there instead. Otherwise, something Afterburn-specific is also fine, like `guestinfo.afterburn.netplan-configs`.

Reply:
`guestinfo.userdata` is for cloud-init userdata (and Ignition could have used that, too). The metadata is meant for platform-provided configuration, not user configuration. The name of the guestinfo field and the content is what VMware "specifies" by reusing what cloud-init already supports, and deviating from to something afterburn-specific would defeat the point - one can already do OS-specific things and this was proposed as an OS-independent way of network configuration. It's not so nice that cloud-init defines this in the end but at least netplan is not directly tied to cloud-init. @akutz can maybe give more background.

Reply:
Ahhh sorry, I missed the `user` vs `meta` distinction. OK, sounds fine to me.

Reply:
OK right, I totally missed that `guestinfo.metadata` is used today on VMware + cloud-init to pass Netplan configs. That makes a lot more sense now.